sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -1445,4 +1445,10 @@ private[spark] object QueryCompilationErrors {
   def operationNotSupportPartitioningError(operation: String): Throwable = {
     new AnalysisException(s"'$operation' does not support partitioning")
   }
+
+  def mixedRefInAggFunc(a: Expression): Throwable = {
+    val msg = "Found an aggregate function in a correlated predicate that has both " +
+      "outer and local references, which is not supported: " + a.sql
```

Comment:
Yea, this message looks clearer than the previous one.

Reply:
`aggregate fucntion` -> `aggregate expression`?

Reply:
I think aggregate fucntion is more user facing